### ISSUE TITLE
Bugs in reference manager

### ISSUE THREAD

**Issue by Alisis33** [radimitrov/CSharpShellApp#123]
There are some bugs in reference manager, all should be tested again.

I found:
- Something is wrong with the project update after a change, it seams csshell forgets added references and errors are shown. They disappeare after reopen the project.
- Adding and deleting references seam to be buggy (or I don't see the mechanism)
- In "Add internal reference" the keyboard is not avalable, so you can not search.

(There are maybe more bugs, or they could be fixed already in a new ubdate.)

**Comment by radimitrov** [comment]
Yes. I've already fixed the forgetful reference issue when adding a new one. 
Deleting a reference is done with long click. But it only works if there is an extra reference you've added. Doesn't work for the defaults like mscorlib. Or the "wireframe icon" references. Those are resolved dependencies.
I'm aware the search isn't opening the keyboard in the internal assemblies, but I'm not sure why. 
It should work... It is the same logic as the search for the referenced ones.
